projects/plugins/jetpack/class.json-api-endpoints.php

```diff
@@ -1451,8 +1451,13 @@ public function get_author( $author, $show_email_and_ip = false ) {
 				$nice       = $user->user_nicename;
 			}
 			if ( defined( 'IS_WPCOM' ) && IS_WPCOM && ! $is_jetpack ) {
-				$active_blog = get_active_blog_for_user( $id );
-				$site_id     = $active_blog->blog_id;
+				$active_blog = apply_filters( 'wpcom_pre_get_active_blog_for_user', false, $id );
+				if ( false === $active_blog ) {
+					$active_blog = get_active_blog_for_user( $id );
+				}
+				if ( ! empty( $active_blog ) ) {
+					$site_id = $active_blog->blog_id;
+				}
```

Comment:
I think we also need to initialize `$site_id` to make sure it's not undefined, especially given that we're now setting it inside the `if` block.
```suggestion
				$site_id = -1;

				$active_blog = apply_filters( 'wpcom_pre_get_active_blog_for_user', false, $id );
				if ( false === $active_blog ) {
					$active_blog = get_active_blog_for_user( $id );
				}
				if ( ! empty( $active_blog ) ) {
					$site_id = $active_blog->blog_id;
				}
```

Reply:
Sounds good. I mistakenly thought the init on line 1394 applies, but it's actually within a different branch.